src/main/java/org/mule/extension/http/internal/listener/HttpListener.java

```diff
@@ -322,6 +331,37 @@ public void handleRequest(HttpRequestContext requestContext, HttpResponseReadyCa
         }
       }
 
+      private String getMethod(HttpRequestContext requestContext) {
+        return requestContext.getRequest().getMethod();
+      }
+
+      private MultiMap<String, String> getHeaders(Result<InputStream, HttpRequestAttributes> result) {
+        return result.getAttributes().get().getHeaders();
```

Comment:
My eyes were hurting too with this. We now extract the method and headers from the _result_. Thanks!